mongo/integration/initial_dns_seedlist_discovery_test.go

```diff
@@ -172,9 +172,13 @@ func getServerByAddress(address string, topo *topology.Topology) (description.Se
 		return []description.Server{}, nil
 	})
 
-	selectedServer, err := topo.SelectServerLegacy(context.Background(), selectByName)
+	selectedServer, err := topo.SelectServer(context.Background(), selectByName)
 	if err != nil {
 		return description.Server{}, err
 	}
-	return selectedServer.Server.Description(), nil
+	selectedServerConnection, err := selectedServer.Connection(context.Background())
+	if err != nil {
+		return description.Server{}, err
+	}
+	return selectedServerConnection.Description(), nil
```

Comment:
You should do `defer selectedServerConnection.Close()` to ensure we're not leaking connections.

Reply:
Ah gotcha. For some reason I didn't think `selectedServer.Connection(...)` was actually opening a connection 🤦 .